docs/development/workflow/development_workflow.rst

```diff
@@ -428,6 +428,22 @@ You may be asked to make changes in the discussion of the pull request. Make
 those changes in your local copy, commit them to your local repo and push them
 to GitHub. GitHub will automatically update your pull request.
 
+.. _no git pull:
+
+Do not create a merge commit
+****************************
+
+If your branch associated with the pull request falls behind the ``master``
+branch of https://github.com/astropy/astropy , GitHub might offer you option
+to catch up or resolve conflict via its web interface; do not use them. Using
+them might create a "merge commit" in your commit history, which is
+undesirable, as a "merge commit" can introduce maintenance overhead for the
+release manager. Do not use ``git pull`` command either.
```

Comment:
"and mess up the branch structure". Not sure how to say that in a document friendly way.

Reply:
Me neither. A typical contributor might not care about preserving branch structure anyway, but I think we need to convey the *impact* of such a thing happening. Suggestions welcome!

Reply:
Maybe: `a "merge commit" introduces undesirable branch structure complexity.` @eteq might want to chime in.
Bottom line is that one merge commit is not the end of the word, but usually it escalates and having two usually means that the PR became a spagetti mess, and not rebasable any  more only with much larger effort. Keeping things clean and tidy is not a bad policy.

Reply:
@bsipocz , your suggestion was added as part of copyediting by @lglattly . FYI. Thank you both!